### DISCUSSION TITLE
KMQTT and KMM

### DISCUSSION THREAD

**Discussion by loves-bruins** [davidepianca98/KMQTT#3] [Q&A] [answered]
Hello,
Can I leverage KMQTT via KMM (Android/iOS)? KMM is Kotlin Multiplatform Mobile

**Comment by davidepianca98** [comment]
> Hello,
> Can I leverage KMQTT via KMM (Android/iOS)? KMM is Kotlin Multiplatform Mobile

Yes you can, but it might not be what you need as KMQTT is a server implementation, while usually on mobile devices the client is used to connect to a MQTT server

**Reply by loves-bruins** [reply]
You are correct, David!  I do need an MQTT client.  Do you know of any KMM MQTT clients offhand?

**Reply by davidepianca98** [reply]
I don't think there exists one yet, which can compile to native for iOS, but I might be wrong. Another solution might be to use Paho Java on Android and a different library for iOS and implement a common interface

**Comment by davidepianca98** [comment, marked as answer]
KMQTT now provides a multiplatform client implementation!
